src/core/lib/channel/channelz_registry.cc

```diff
@@ -58,19 +45,24 @@ ChannelzRegistry* ChannelzRegistry::Default() {
   return g_channelz_registry;
 }
 
-ChannelzRegistry::ChannelzRegistry() : uuid_(1) {
+ChannelzRegistry::ChannelzRegistry() {
   gpr_mu_init(&mu_);
-  avl_ = grpc_avl_create(&avl_vtable);
+  // init elements to nullptr
+  memset(entities_.data(), 0, sizeof(void*) * entities_.capacity());
 }
 
 ChannelzRegistry::~ChannelzRegistry() {
-  grpc_avl_unref(avl_, nullptr);
+  // null out all elements so that the destructors don't try to run.
```

Comment:
There's no desctructor to run for `void*`.  And if there were, setting the memory to zero wouldn't prevent them from running.

Reply:
Hmm, this fixed a crash before. But now it works. Must have been lucky, then I fixed the crash elsewhere. Done